filtering-stories-by-status-dryml

From: Bryan Larsen <bryan@larsen.st>

## Filtering stories by status

Rapid's `<table-plus>` is giving us some nice searching and sorting features on the project page. We can easily add some filtering into the mix, so that it's easy to, say, see only new stories.

First we'll add the filter control to the header of the table-plus. Rapid provides a `<filter-menu>` tag which is just what we need. We want to add it to the header section, before the stuff that's already there. In DRYML, you can prepend or append content to any named parameter. `<table-plus>` has a `header:` parameter, so we can use `<prepend-header:>`, like this:

SHOW_PATCH
---
 app/views/projects/show.dryml |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/app/views/projects/show.dryml b/app/views/projects/show.dryml
index 0ea6a1d..e79d1d0 100644
--- a/app/views/projects/show.dryml
+++ b/app/views/projects/show.dryml
@@ -1,6 +1,11 @@
 <show-page>
   <collection: replace>
   <table-plus with="&@stories" fields="this, tasks_count, status">
+    <prepend-header:>
+      <div class="filter">
+        Display by status: <filter-menu param-name="status" options="&StoryStatus.all"/>
+      </div>
+    </prepend-header:>
     <empty-message:>No stories match your criteria</empty-message:>
   </table-plus>
   </collection:>
